src/components/TableView.tsx

```diff
@@ -0,0 +1,122 @@
+import React from "react";
+import {TableColumnData, TableData, TableRowData} from "../api/Interfaces";
+import {Card, Collapse, Table} from "react-bootstrap";
+import TableRowView from "./TableRowView";
+import TableColumnTogglesView from "./TableColumnTogglesView";
+import ToggleColumnsButton from "./ToggleColumnsButton";
+
+interface TableViewProps {
+    table: TableData
+}
+
+interface TableViewState {
+    cols: TableColumnData[],
+    toggleColumnsExpanded: boolean
+}
+
+class TableView extends React.Component<TableViewProps> {
+    state: TableViewState;
+
+    constructor(props: TableViewProps) {
+        super(props);
+
+        this.state = {
+            cols: this.cloneShallow(this.props.table.columns),
+            toggleColumnsExpanded: false,
+        };
+    }
+
+    cloneShallow(cols: TableColumnData[]) {
+        // clone required to not overwrite the original state for resetting
+        let result: TableColumnData[] = [];
+        cols.forEach(c => {
+            result.push({
+                id: c.id,
+                name: c.name,
+                key: c.key,
+                description: c.description,
+                column_id: c.column_id,
+                table_id: c.table_id,
+                column_type: c.column_type,
+                order: c.order,
+                hidden: c.hidden
+            } as TableColumnData)
+        });
+        return result;
+    }
+
+    toggleColumn(cols: TableColumnData[]) {
+        this.setState({cols: cols});
```

Comment:
Codacy has a fix for the issue: [Expected property shorthand.](https://app.codacy.com/gh/ranimepiracy/index/pullRequest?prid=7272556)
```suggestion
        this.setState({cols});
```